{"pr_number": 2220, "pr_title": "Fix for Issue #1684 Changes in handleTimeoutCausedException in Timeou\u2026", "pr_createdAt": "2020-12-25T00:29:07Z", "pr_url": "https://github.com/aws/aws-sdk-java-v2/pull/2220", "timeline": [{"id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDU1MDM0NjUxOA==", "url": "https://github.com/aws/aws-sdk-java-v2/pull/2220#discussion_r550346518", "bodyText": "We check isCausedByApiCallAttemptTimeout but can this be caused by the API call timer as well?", "author": "dagnir", "createdAt": "2020-12-30T22:37:11Z", "path": "core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/TimeoutExceptionHandlingStage.java", "diffHunk": "@@ -90,7 +91,8 @@ public TimeoutExceptionHandlingStage(HttpClientDependencies dependencies, Reques\n      */\n     private Exception translatePipelineException(RequestExecutionContext context, Exception e) {\n         if (e instanceof InterruptedException || e instanceof IOException ||\n-            e instanceof AbortedException || Thread.currentThread().isInterrupted()) {\n+                e instanceof AbortedException || Thread.currentThread().isInterrupted()\n+                || (e instanceof SdkClientException && isCausedByApiCallAttemptTimeout(context))) {", "originalCommit": "209764f246a19185a65e51dc6dd60c1b9d110ba0", "replyToReviewId": null, "replies": [{"id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDU1MDM4MjY3Ng==", "url": "https://github.com/aws/aws-sdk-java-v2/pull/2220#discussion_r550382676", "bodyText": "The timeout exception is caused while  ApiCallAttemptTimeoutExecute-Link and this is the same check which resets the interrupts\nApiCallAttemptTimeoutTrackingStageInterruptReset-Link.\nAPI call timer  causes context.apiCallTimeoutTracker() while the one used above is\ncontext.apiCallAttemptTimeoutTracker()", "author": "joviegas", "createdAt": "2020-12-31T02:56:02Z", "replyToReviewId": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDU1MDM0NjUxOA=="}, {"id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDU1MDY0Nzc4Mg==", "url": "https://github.com/aws/aws-sdk-java-v2/pull/2220#discussion_r550647782", "bodyText": "Okay gotcha, so if the API call timer fired (and not the attempt timer), then the exception would be AbortedException which we already handle.", "author": "dagnir", "createdAt": "2020-12-31T18:56:41Z", "replyToReviewId": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDU1MDM0NjUxOA=="}], "type": "inlineReview", "revised_code": null}, {"oid": "518683e2dc919d3bd2c2528c661bda7faabb6c7d", "url": "https://github.com/aws/aws-sdk-java-v2/commit/518683e2dc919d3bd2c2528c661bda7faabb6c7d", "message": "Fix for Issue #1684 Changes in handleTimeoutCausedException in TimeoutExceptionHandlingStage", "committedDate": "2021-01-04T17:58:02Z", "type": "commit"}]}